packages/component-base/test/list-mixin.test.js

```diff
@@ -4,217 +4,220 @@ import {
+          display: none !important;
+        }
+
+        :host(.hidden-attribute) {
+          display: none;
+        }
+      </style>
+      <slot></slot>
+    `,
+    (Base) =>
+      class extends baseMixin(Base) {
+        static get properties() {
+          return {
+            disabled: {
+              type: Boolean,
+              reflectToAttribute: true,
+              observer: '_disabledChanged',
+            },
+            focused: {
+              type: Boolean,
+            },
+            selected: {
+              type: Boolean,
+            },
+          };
+        }
+
+        constructor() {
+          super();
+
+          this._hasVaadinItemMixin = true;
+        }
+
+        _disabledChanged(disabled) {
+          if (disabled) {
+            // Mimic the ItemMixin logic
+            this.selected = false;
           }
+        }
+      },
+  );
 
-          :host([orientation='vertical']) #scroll {
-            height: 100%;
-            flex-direction: column;
-          }
-        </style>
-        <div id="scroll">
-          <slot></slot>
-        </div>
-      `;
-    }
-
-    get _scrollerElement() {
-      return this.$.scroll;
-    }
-  },
-);
-
-customElements.define(
-  'test-item-element',
-  class extends PolymerElement {
-    static get template() {
-      return html`
-        <style>
-          :host {
-            display: block;
-          }
+  describe('items', () => {
+    beforeEach(async () => {
+      list = fixtureSync(`
+        <${listTag}>
+          <${itemTag}>Item 0</${itemTag}>
+          <${itemTag}>Item 1</${itemTag}>
+          <hr />
+          <${itemTag}>Item 2</${itemTag}>
+          <${itemTag}>Item 3</${itemTag}>
+          <hr />
+          <${itemTag}>Item 4</${itemTag}>
+        </${listTag}>
+      `);
+      await nextRender();
+    });
 
-          :host([hidden]) {
-            display: none !important;
-          }
+    it('should set items based on the children count', () => {
+      expect(list.items.length).to.be.equal(5);
+    });
 
-          :host(.hidden-attribute) {
-            display: none;
-          }
-        </style>
-        <slot></slot>
-      `;
-    }
-
-    static get properties() {
-      return {
-        _hasVaadinItemMixin: {
-          value: true,
-        },
-        disabled: {
-          type: Boolean,
-          reflectToAttribute: true,
-          observer: '_disabledChanged',
-        },
-        selected: {
-          type: Boolean,
-        },
-      };
-    }
-
-    _disabledChanged(disabled) {
-      if (disabled) {
-        // Simplified version of Vaadin.ItemMixin behavior
-        this.selected = false;
-      }
-    }
-  },
-);
-
-customElements.define(
-  'test-list-wrapper-element',
-  class extends PolymerElement {
-    static get template() {
-      return html`
-        <style>
-          :host {
-            display: block;
-          }
-        </style>
+    it('should update items list when adding nodes', async () => {
```

Comment:
```suggestion
    it('should update items when adding nodes', async () => {
```

Reply:
Updated this test and two others accordingly.